pageserver/src/layered_repository/filename.rs

```diff
@@ -35,7 +37,7 @@ impl DeltaFileName {
     /// Parse a string as a delta file name. Returns None if the filename does not
     /// match the expected pattern.
     ///
-    pub fn from_str(fname: &str) -> Option<Self> {
+    pub fn parse_str(fname: &str) -> Option<Self> {
```

Comment:
Clippy appealed that `from_str` might confuse people and make them think about `FromStr`.